haas/ext/switches/_console.py

```diff
@@ -74,6 +74,11 @@ def disconnect(self):
         """End the session. Must be at the main prompt."""
 
     def _set_native(self, old_native, network_id, interface):
+        """This method is intended to be over written.
+
+        In some switches, we need to create a vlan before setting that as
+        native. `interface` is needed to get back to the interface prompt
+        to set the native vlan for that interface"""
```

Comment:
This smells funky to me. You're basically shoving the whole implementation in the docstring; there's got to be a cleaner way to define/describe the interface.